src/middleware/routing.ts

```diff
@@ -1,22 +1,24 @@
-import Router, { RouterContext } from '@koa/router';
+import Router, { Middleware, RouterParamContext } from '@koa/router';
 import createHttpError from 'http-errors';
-import { Middleware, Next, DefaultState } from 'koa';
+import { DefaultContextExtends, DefaultStateExtends, Next } from 'koa';
 import compose from 'koa-compose';
 
-const notFound = (): Middleware<DefaultState, RouterContext> => (
-  async ({ _matchedRoute }: RouterContext, next: Next): Promise<void> => {
-    await next();
+const notFound = <State extends DefaultStateExtends, Context extends DefaultContextExtends>
+  (): Middleware<State, Context> => (
+    async ({ _matchedRoute }: RouterParamContext<State, Context>, next: Next): Promise<void> => {
```

Comment:
yes, this is weird indentation; it changes depending on how the previous line is broken up?

Reply:
Seems to be a difference between an assignment and an export...